homeassistant/components/nut/config_flow.py

```diff
@@ -59,15 +61,15 @@ async def validate_input(hass: core.HomeAssistant, data):
     username = data.get(CONF_USERNAME)
     password = data.get(CONF_PASSWORD)
 
-    data = PyNUTData(host, port, alias, username, password)
-    await hass.async_add_executor_job(data.update)
-    if not (status := data.status):
+    nut_data = PyNUTData(host, port, alias, username, password)
+    await hass.async_add_executor_job(nut_data.update)
+    if not (status := nut_data.status):
         raise CannotConnect
 
-    return {"ups_list": data.ups_list, "available_resources": status}
+    return {"ups_list": nut_data.ups_list, "available_resources": status}
 
 
-def _format_host_port_alias(user_input):
+def _format_host_port_alias(user_input: dict[str, Any]) -> str:
```

Comment:
We can use `Mapping` here instead of `dict`. Then we don't need to copy the config entry data before passing it here.